src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,108 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final int DEFAULT_SIZE = 0;
+    private T removedValue;
+    private Object[] list;
+    private int size;
+
+    public ArrayList() {
+        list = new Object[DEFAULT_CAPACITY];
+        size = DEFAULT_SIZE;
+    }
+
+    public Object[] grow(int size) {
+        return list = Arrays.copyOf(list, (size + size / 2));
+    }
 
     @Override
     public void add(T value) {
-
+        if (size >= list.length) {
+            list = grow(size);
```

Comment:
you are actually assigning the result to list twice - here and inside `grow()` method